src/NuGet.Clients/NuGet.CommandLine/Commands/PushCommand.cs

```diff
@@ -51,6 +51,11 @@ public class PushCommand : Command
                     NoSymbols,
                     Console);
             }
+            catch (TaskCanceledException ex)
+            {
+                string timeoutMessage = LocalizedResourceManager.GetString(nameof(NuGetResources.PushCommandTimeoutError));
+                throw new AggregateException(ex, new Exception(timeoutMessage));
```

Comment:
Yeah the console programs catch all exceptions and print out their message as an error. I used AggregateException so that the original TaskCanceledException can be preserved to show its message and optionally its callstack (with verbose output is turned on). And then I combined it with a generic Exception so that a custom message can be displayed.

But if the callstack of the original TaskCanceledException doesn't matter, I can just throw a new specific type of exception here.

---
In reply to: [58240759](https://github.com/NuGet/NuGet.Client/pull/442#discussion_r58240759) [](ancestors = 58240759)